src/RobotInterface/YarpImplementation/src/YarpSensorBridge.cpp

```diff
@@ -0,0 +1,440 @@
+}
+
+bool YarpSensorBridge::getIMUsList(std::vector<std::string>& IMUsList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getIMUsList]"))
+    {
+        return false;
+    }
+    IMUsList = m_pimpl->metaData.sensorsList.IMUsList;
+    return true;
+}
+
+bool YarpSensorBridge::getLinearAccelerometersList(std::vector<std::string>& linearAccelerometersList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getLinearAccelerometersList]"))
+    {
+        return false;
+    }
+    linearAccelerometersList = m_pimpl->metaData.sensorsList.linearAccelerometersList;
+    return true;
+}
+
+bool YarpSensorBridge::getGyroscopesList(std::vector<std::string>& gyroscopesList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getGyroscopesList]"))
+    {
+        return false;
+    }
+    gyroscopesList = m_pimpl->metaData.sensorsList.gyroscopesList;
+    return true;
+}
+
+bool YarpSensorBridge::getOrientationSensorsList(std::vector<std::string>& orientationSensorsList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getOrientationSensorsList]"))
+    {
+        return false;
+    }
+    orientationSensorsList = m_pimpl->metaData.sensorsList.orientationSensorsList;
+    return true;
+}
+
+bool YarpSensorBridge::getMagnetometersList(std::vector<std::string>& magnetometersList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getMagnetometersList]"))
+    {
+        return false;
+    }
+    magnetometersList = m_pimpl->metaData.sensorsList.magnetometersList;
+    return true;
+}
+
+bool YarpSensorBridge::getSixAxisForceTorqueSensorsList(std::vector<std::string>& sixAxisForceTorqueSensorsList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getSixAxisForceTorqueSensorsList]"))
+    {
+        return false;
+    }
+    sixAxisForceTorqueSensorsList = m_pimpl->metaData.sensorsList.sixAxisForceTorqueSensorsList;
+    return true;
+}
+
+bool YarpSensorBridge::getCartesianWrenchesList(std::vector<std::string>& cartesianWrenchesList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getCartesianWrenchesList]"))
+    {
+        return false;
+    }
+    cartesianWrenchesList = m_pimpl->metaData.sensorsList.cartesianWrenchesList;
+    return true;
+}
+
+bool YarpSensorBridge::getRGBCamerasList(std::vector<std::string>& rgbCamerasList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getRGBCamerasList]"))
+    {
+        return false;
+    }
+    rgbCamerasList = m_pimpl->metaData.sensorsList.rgbCamerasList;
+    return true;
+}
+
+bool YarpSensorBridge::getRGBDCamerasList(std::vector<std::string>& rgbdCamerasList)
+{
+    if (!m_pimpl->checkValid("[YarpSensorBridge::getRGBDCamerasList]"))
+    {
+        return false;
+    }
+    rgbdCamerasList = m_pimpl->metaData.sensorsList.rgbdCamerasList;
+    return true;
+}
+
+bool YarpSensorBridge::getJointPosition(const std::string& jointName,
+                                        double& jointPosition,
+                                        double* receiveTimeInSeconds)
```

Comment:
I'm missing something for sure, but is there a reason why you chose to use the pointer instead of a reference? (The same comment is valid for the others methods)

Reply:
To make it optional.. in the headers it's defined with a default `nullptr` assignment.
This was suggested by @S-Dafarra because some times, we are not interested in timestamps, so we can just directly get the measurement.